### PR TITLE
Algorithms appended

### PR DESCRIPTION
**Pull request by abhishekslab**
All the c++ programs are based on various algorithms of Data Structures including algorithms like conversion of an infix expression to postfix expression using stack which comes under application of stack and various other topics such as queue through array, stack via linked list etc.

### CODE REVIEW

File: src/LIST.CPP [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/*------------------------------------------------------------------------*/
+/*                                                                        */
+/*  LIST.CPP                                                              */
+/*                                                                        */
+/*  Copyright Borland International 1991                                  */
```
Comment (holmgr) [inline src/LIST.CPP:5]: Is this really your code? I do not want to accept code which has been taken with out consent.

File: src/LinkedList.CPP [hunk 2]
```diff
@@ -0,0 +1,205 @@
+#include<stdio.h>
+#include<conio.h>
+#include<alloc.h>
+#include<stdlib.h>
+
+//malloc fdunction is generic
+//append insert at end awa first node
```
Comment (holmgr) [inline src/LinkedList.CPP:7]: What are these comments for exactly? Which struct, or function do they belong to?

File: src/Stack.CPP [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*------------------------------------------------------------------------*/
+/*                                                                        */
+/*  STACK.CPP                                                             */
+/*                                                                        */
+/*  Copyright Borland International 1991                                  */
```
Comment (holmgr) [inline src/Stack.CPP:5]: s this really your code? I do not want to accept code which has been taken without consent.

File: src/SwapList.CPP [hunk 4]
```diff
@@ -0,0 +1,242 @@
+//Swapping adjacent nodes in single linked list
+
+#include<stdio.h>
+#include<conio.h>
+#include<alloc.h>
+#include<stdlib.h>
+
+//malloc fdunction is generic
+//append insert at end awa first node
```
Comment (holmgr) [inline src/SwapList.CPP:9]: What are these comments referring to?

**Review by holmgr — COMMENTED** [review]
Hi. Sorry if this comes across strongly but I want to make it clear.
Firstly, you should really look into proper indentation and code formatting so that your code becomes readable. I would never accept code this badly formatted/unindented in any project which I have any control over. Secondly, you take input in many of your methods which is not very generic and usable except for demonstration purposes. It would require quite a large rewrite to actually use this. Thirdly, you lack overall documentation for the different methods.
Fourthly, you seem to have taken someone else's code (since it is copyrighted by some company), if correct this is stealing and **not** a good behavior! Finally, this project is kind of on the backburner and therefor I will not have time to help your to clean it up. 
Ps. I suspect this is just a PR to try to get a free shirt in Hacktober...

**Comment by abhishekslab** [comment]
Yeah ,I am sorry for the various flaws in my PR, I am a beginner at git hub and open source .

In src/Stack.CPP<https://github.com/holmgr/dsa-cozy-corner/pull/14#discussion_r225795401>:

> @@ -0,0 +1,37 @@

+/*------------------------------------------------------------------------*/

+/*                                                                        */

+/*  STACK.CPP                                                             */

+/*                                                                        */

+/*  Copyright Borland International 1991                                  */

s this really your code? I do not want to accept code which has been taken without consent.
The above one  actually is a copyrighted thing. I was actually looking on a page for help related to some concepts over the above program and I referenced it to clear up the wrong code I was writing, we all learn from somewhere?. And I by mistake thought it to be my version of the same program I was writing and appended it.


In src/LinkedList.CPP<https://github.com/holmgr/dsa-cozy-corner/pull/14#discussion_r225795312>:

> @@ -0,0 +1,205 @@

+#include<stdio.h>

+#include<conio.h>

+#include<alloc.h>

+#include<stdlib.h>

+

+//malloc fdunction is generic

+//append insert at end awa first node

What are these comments for exactly? Which struct, or function do they belong to?
For the above and rest all are the programs submitted were written by me (I rechecked them all) in the past week and the above comments were done by me to get to remember the concept of memory allocation called by the malloc function in c++ under the alloc.h / stdlib.h  library. I am and was practising and keeping the key points to remember in the comments.( Maybe you are not recalling the fact that malloc is a generic function in c++ thus the value returned by it needs to be typecasted for the structure you want to allocate it’s memory to)

"Firstly, you should really look into proper indentation and code formatting so that your code becomes readable. I would never accept this code this badly formatted/unindented in any project which have any control over. Secondly, you take input in many of your methods which is not very generic and usable except for demonstration purposes"

Thanks for the above lines as they are a means of feedback for my style and the way I write code and I would really get indulgd in improving them.

"Finally, this project is kind of on the backburner and therefor I will not have time to help your to clean it up.
Ps. I suspect this is just a PR to try to get a free shirt in Hacktober..."

Well the above two statements were actually harsh , and yep I am interested in getting that T-shirt as a momento for my beginning in open source and I would have loved if you had actually told me what to correct as all the code maybe for you was a backburner but actually took a lot of effort from my side to write it up and I spended my whole last week in understanding the concepts and writing them.

**Comment by holmgr** [comment]
Yes, sorry if I came across hard, but I also need to be clear in my stance.

This is a backburner project for me, which is quite clear since the last commit was over a year ago, which is a good thing to check before you contribute to projects in the future. The fact that I actually responded at all is taking time out of my quite busy day to give you some type of feedback, which is actually more than what you probably would get in most inactive projects.
I do get why you want to get that shirt (I also want one) but before I submit patches to other peoples repositories I would first check what types of submissions are wanted (are there issues, or a part in the readme which contains any suggestions?). 

Then I would start by doing a small implementation/fix (not a massive task right away, or at least split my submission into different PRs), or just ask in an issue if I could work on that problem (all to reduce the possibility of wasting my time) .

Then I would check that my code conforms to what the maintainer wants by either checking the readme/contribution guidelines if there are any (there is one for this project https://github.com/holmgr/dsa-cozy-corner/blob/master/CONTRIBUTING.md) or by again asking in an issue, and **then but only** then submit at PR to get the highest chance of acceptance and to use the minimal effort from the maintainer.

The fact that you have not seemed to have read or followed any of the conventions of the existing code or the explicit guidelines makes me think that this is a waste of time for me, and therefor the quick and quite harsh comments. If you have not read my readme/guidelines, why should I read and give feedback on your code?

Also the Copyright comments are really a red flag for anyone, and if you truly wrote the code yourself than why did you copy those bits? This makes me a bit scared to accept any of the code in fact since I could be liable for a copyright claim.

So, for future PRs to this or other projects, here is some abbreviated tips:
* Firstly get to know what is wanted, do not be afraid to ask in an issue
* Then implement a minor addition or bug fix
* Make sure that the code change works by writing tests if the project requires it
* Make sure that the new/changed code follows the styleguide/conventions (implicitly by checking the current codebase, or by checking the explicit guidelines in the project?
* Then submit a PR with the change.

All these steps are to make sure that you as a contributor wastes as little of your time as possible and secondly wastes as little of the maintainers time as possible.

**Comment by abhishekslab** [comment]
Thanks for the feedback and I will remember the above-listed points and come up with an improved PR next time, I am currently working on the terms you have used inside your code and will come up with a better patch soon.
                              Thanks for the guiding part and being harsh can sometimes actually trigger someone in the right direction :)

**Pull request closed without merging**